codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/generators/protocol/ServerProtocolTestGenerator.kt

```diff
@@ -522,21 +658,24 @@ class ServerProtocolTestGenerator(
         }
     }
 
-    private fun checkHttpOperationExtension(rustWriter: RustWriter) {
-        rustWriter.rustTemplate(
-            """
-            let operation_extension = http_response.extensions()
-                .get::<#{SmithyHttpServer}::extension::OperationExtension>()
-                .expect("extension `OperationExtension` not found");
-            """.trimIndent(),
-            *codegenScope,
-        )
-        rustWriter.writeWithNoFormatting(
-            """
-            assert_eq!(operation_extension.absolute(), format!("{}.{}", "${operationShape.id.namespace}", "${operationSymbol.name}"));
-            """.trimIndent(),
-        )
-    }
+    // We can't check that the `OperationExtension` is set in the response, because it is set in the implementation
+    // of the operation `Handler` trait, a code path that does not get exercised when we don't have a request to
+    // invoke it with (like in the case of an `httpResponseTest` test case).
```

Comment:
We could perform this check _after_ invoking the router with the request right? That is, above we construct:

```
let res = router
    .call(http_request)
    .await
    .expect("unable to make an HTTP request");

// Check that the extension is set on the response here.
```

Reply:
Addressed with <https://github.com/awslabs/smithy-rs/pull/1708/commits/440dbae47be6be45119bf028112aa6b42cecd076>, though [this particular case](https://github.com/awslabs/smithy/blob/f598b87c51af5943686e38706847a5091fe718da/smithy-aws-protocol-tests/model/restJson1/http-labels.smithy#L254-L266) failed as it cannot find any extensions from the HTTP response. The HTTP status Code is a 404.

```
[rest_json/rust-server-codegen/src/operation.rs:877] http_response = Response {
    status: 404,
    version: HTTP/1.1,
    headers: {
        "content-type": "application/json",
        "x-amzn-errortype": "UnknownOperationException",
    },
    body: UnsyncBoxBody,
}
```

Haven't dug into details, but it might be a routing bug or something else.

Reply:
Can we take out the `OperationExtension` assertion entirely? 

1. It's inclusion is not part of the Smithy spec
2. It's not added in the deserialization error path
3. It's cannot be added in the routing failure error path
4. The panic from the `todo!`s ([and later `InternalFailure` + 500](https://github.com/awslabs/smithy-rs/pull/1709#issuecomment-1239186367)) signal that tests have routed to an unexpected operation.

Reply:
It's a harmless thing to assert. If the request was processed successfully (which all of the `httpRequestTests` should be), we expect the `OperationExtension` to be there.

Reply:
I guess it signals that we **have** entered a specific operation.

An alternative, if we want to assert that a specific operation is hit then perhaps we can pass a sending end of a channel to the handler closure and then `assert!` we receive a signal at the end of the test.

In both approaches we should log the `http_response` so we get some insight into the exact error.

Reply:
> we should log the http_response so we get some insight into the exact error when deserialization/routing fails.

Yeah we should `debug!()` the response and the request.

Reply:
With the latest two commits 2a5768c1b9671c43b911127a1cd1eac4c67f31cd 334269383fcfdfc42f332f10ae866a20929ab32d and #1719 from Harry, most of the issues of broken protocol tests have been fixed. Before https://github.com/awslabs/smithy/pull/1392 gets merged, we can proceed with temporary fixes.